credentials/credentials.go

```diff
@@ -50,6 +51,43 @@ type PerRPCCredentials interface {
 	RequireTransportSecurity() bool
 }
 
+// SecurityLevel defines the protection level on an established connection.
+type SecurityLevel int
+
+const (
+	// NoSecurity indicates a connection is insecure.
+	NoSecurity SecurityLevel = iota + 1
```

Comment:
Comment: `// The zero SecurityLevel value is invalid for backward compatibility` ?

Reply:
Done.